src/docs/reference/core/KeyVault.Azure/README.md

```diff
@@ -0,0 +1,40 @@
+# Azure Key Vault (`OrchardCore.Azure.KeyVault`)
+The Azure Key Vault configuration provider adds app configuration values from the Azure Key Vault in order to safeguard your cryptographic keys and secrets used by your app. It also contains custom override of the DefaultKeyVaultManger class that retrieves secrets from Azure Key Vault and translates ---
+to an underscore (_)  and -- to a colon (:). Both underscores and colons are illegal characters in Azure KeyVault.
+
+Example:
+Key Vault Input: "OrchardCore--OrchardCore---Shells---Database--ConnectionString".
+Output: "OrchardCore:OrchardCore_Shells_Database:ConnectionString".
+See https://github.com/OrchardCMS/OrchardCore/issues/6359.
+
+## Configuration
```

Comment:
Can we have a small example that uses directly `AzureKeyVaultSecretManager` such that standard ASP.NET tutorials can be followrd?

Reply:
Do you mean an example of IConfiguration? I have been using it to retrieve my media and database connection strings. For example in DatabaseShellConfigurationSources.cs:
![image](https://user-images.githubusercontent.com/22512098/86422868-51250980-bcac-11ea-83ab-684cf8bf0102.png)

Using the following Keys:
OrchardCore--OrchardCore---Shells---Database--ConnectionString
OrchardCore--OrchardCore---Shells---Database--DatabaseProvider (Not actually a secret, but required to get the section)

Reply:
What Seb was asking for is something that matches the getting started guides from the ASP.NET docs here https://docs.microsoft.com/en-us/aspnet/core/security/key-vault-configuration?view=aspnetcore-3.1

Where they are using the `AzureServiceTokenProvider` callbacks and such.

Which makes me also wonder if we should not be using the callback technique, rather than having our own configuration secrets for it?

Reply:
@deanmarcussen 

Here @JoshLefebvre just override the `DefaultKeyVaultSecretManager` with a new `AzureKeyVaultSecretManager` and provide an `AddOrchardCoreAzureKeyVault()` extension that uses it and other config values.

But people can just follow the aspnet documentation and then use `new AzureKeyVaultSecretManager()` in place of `new DefaultKeyVaultSecretManager()`

So maybe just remove the `AddOrchardCoreAzureKeyVault()` that is just an example on how to use the new `AzureKeyVaultSecretManager` and let people follow the doc and do their own extension if they want.

Reply:
Do I understand your implementation right that you have added replacement of `---` to `_` along with default `--` to `:` behavior?

Reply:
@xperiandri  Exactly, this is what @JoshLefebvre did ;)

Reply:
Is that something that is common? Either in OC or ASP.NET with Keyvault?